lab_1_classify_by_unigrams/main.py

```diff
@@ -23,35 +31,65 @@ def tokenize(text: str) -> list[str] | None:
 
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
-    """
-    Calculate frequencies of given tokens.
+    if isinstance(tokens, list):
+        if isinstance(tokens[0], str):
+            frequency = {}
+            tokens_number = len(tokens)
+            for letter in tokens:
+                if isinstance(letter, str):
+                    letter_number = tokens.count(letter)
+                    letter_index = letter_number/tokens_number
```

Comment:
there are a lot of variables that are used only once

Reply:
don't understand

Reply:
for example, letter_number variable is created and used further only in letter_index = letter_number/tokens_number, while it can be combined